Sources/o/URL/o+url.swift

```diff
@@ -9,97 +9,181 @@ import Foundation
+        ) -> URLRequest {
+            var request = URLRequest(url: url)
+
+            request.httpMethod = method.rawValue
+            request.allHTTPHeaderFields = headerFields
+
+            return request
+        }
+    }
 }
 
+// MARK: - Async/ Await
+
 public extension o.url {
-    /// GET
-    static func `in`<Value: Decodable>(
+    static func `get`(
         url: URL,
         headerFields: [String: String] = [
             "Content-Type": "application/json; charset=utf-8"
-        ],
-        successHandler: @escaping (Value, URLResponse) -> Void,
-        errorHandler: @escaping (Error) -> Void = {
-            o.console.out($0.localizedDescription)
-        },
-        noResponseHandler: @escaping () -> Void = {
-            o.console.out("No URL Response")
-        },
-        failureHandler: @escaping (URLResponse) -> Void = {
-            o.console.out("No Data (\(dump($0)))")
-        },
-        decodingErrorHandler: @escaping (Error) -> Void = {
-            o.console.out($0.localizedDescription)
-        }
-    ) {
-        var request = URLRequest(url: url)
-        
-        request.httpMethod = "GET"
-        request.allHTTPHeaderFields = headerFields
-        
-        URLSession.shared
-            .dataTask(
-                with: request,
-                completionHandler: { data, response, error in
-                    if let error = error {
-                        errorHandler(error)
-                    }
-                    
-                    guard let response = response else {
-                        noResponseHandler()
-                        return
-                    }
-                    
-                    guard let data = data else {
-                        failureHandler(response)
-                        return
-                    }
-                    
-                    if let data = data as? Value {
-                        successHandler(data, response)
-                        return
-                    }
-                    
-                    do {
-                        let value = try JSONDecoder()
-                            .decode(Value.self, from: data)
-                        successHandler(value, response)
-                    } catch {
-                        decodingErrorHandler(error)
-                    }
-                }
+        ]
+    ) async throws -> (Data?, URLResponse?) {
+        try await URLSession.shared.data(
+            for: request(
+                for: url,
+                method: .GET,
+                headerFields: headerFields
+            )
+        )
+    }
+
+    static func head(
+        url: URL,
+        headerFields: [String: String]
+    ) async throws -> URLResponse? {
+        try await URLSession.shared.data(
+            for: request(
+                for: url,
+                method: .HEAD,
+                headerFields: headerFields
             )
-            .resume()
+        ).1
```

Comment:
Wwhat are the `.1`'s? I see them at the end of most of these functions.

Reply:
URLSession data function returns a tuple, `(Data?, URLResponse?)` and `.0` is Data and `.1` is the Response.

Reply:
Some HTTP method shouldn't return Data I believe. So those ones have the `.1` to only return the response.

Reply:
Understood!